### ISSUE TITLE
Thread for possible enhancements in future

### ISSUE THREAD

**Issue by abhi18av** [TORCH-Consortium/MAGMA#120]
On this thread we can keep track of the enhancements which would be nice to have.

**Comment by abhi18av** [comment]
Concerning the Structural Variants: 

1. There is some ongoing work for accommodate an alternative structural variants subworkflow (in addition to the `delly` one) #146 

2. AS we rely heavily on GATK4 for the core of the pipeline, would https://github.com/broadinstitute/gatk-sv be worth exploring?

**Comment by TimHHH** [comment]
>     2. As we rely heavily on GATK4 for the core of the pipeline, would https://github.com/broadinstitute/gatk-sv be worth exploring?
Yes this is very interesting, but this will take some serious hands-on time to implement and and test, not something for in the near future. For now the TBprofiler approach is easier 👍

**Comment by abhi18av** [comment]
Gotcha - thanks for the suggestions Tim 😊 

- [ ] Adding another thought here regarding the `gatk` package - one of the core strengths of the package is its integration with `spark` which can speed up the overall analysis.

Recently, while working on an analysis (on CERI MAC) I kept running across the memory failure on `gatk MarkDuplicates` step even when I was using the `docker` images. On a side note, my suspicion is that it has something to do with docker-desktop's Hyperkit configuration ([see reference here](https://docs.google.com/document/d/17ZiQC1Tp9iH320K-uqVLyiJmk4DHJ3c4zgQetJiKYQM/edit)), which I circumvented by relying on conda integration.

However, this led me to investigate a little bit about the `gatk4-spark` packages - I'd like to explore that later to see if it could help us optimize the pipeline for a setting where we have smaller-yet-multiple compute nodes, especially the case with cloud.

**Comment by abhi18av** [comment]
- [ ] Consider adding spoligotyping either via the original https://anaconda.org/bioconda/spotyping package or the `experimental` support in tb-profiler https://github.com/jodyphelan/TBProfiler#spoligotyping

**Comment by abhi18av** [comment]
We could also rely upon `genozip` https://www.genozip.com to compress the resultant outputs of the pipeline, in the extreme cases that people would like to use this pipeline for further  exploration of VCF/BAM files.
